onnx_tf/backend.py

```diff
@@ -294,7 +294,7 @@ def onnx_graph_to_tensorflow_net(cls, graph_def, opset):
       for value_info in graph_def.input:
         if value_info.name in initialized:
           continue
-        shape = list(d.dim_value if d.dim_value >= 0 else None
+        shape = list(d.dim_value if d.dim_value > 0 else None
```

Comment:
I'm curious why this change?

I personally cannot think of any cases where dimension size should be zero. Thus I think somewhere in the graph definition, a mistake is being made and we should not (potentially prematurely) help the user fix the mistake by converting a zero to a None.

On the other hand, there might be legitimate reasons to change (for instance if ONNX makes it clear that dim size = 0 means any number would fit) that I do not know about. Could you elaborate?

Reply:
Yes, sorry, I should have mentioned this. I have asked the ONNX developers about statically unknown dimension sizes in [this thread](https://github.com/onnx/onnx/issues/740), and the way to specify this is to use the `dim_param` attribute. When you do this, the value of `dim_value` automatically becomes 0, so I figured that a 0 represents a dimension of arbitrary size. However, it might be a good idea to ask the developers about this, since this does not appear in the specification (I think).

Reply:
From your description, using dim_param is **not a necessary condition** for dim_value to be 0. In other words, dim_value=0 does not imply that the user used dim_param. There are still possibilities that users made a mistake. 

So I would suggest you pull an update from the master branch (I think something is being done to check dim_param). And there the corresponding contributor check the root cause (aka dim_param), not a manifestation (aka dim_value) of the problem. Please take a look and see if the update resolves your concern.

Reply:
You are right, the solution from the master branch solves the problem of underspecified dimensions in a much cleaner way. Thank you.